packages/core/EventManager.ts

```diff
@@ -970,8 +974,26 @@ export default class EventManager {
 
   private async createAllCRMEvents(event: CalendarEvent) {
     const createdEvents = [];
+
+    const featureRepo = new FeaturesRepository();
+    const isTaskerEnabledForSalesforceCrm = event.team?.id
+      ? await featureRepo.checkIfTeamHasFeature(event.team.id, "salesforce-crm-tasker")
+      : false;
+
     const uid = getUid(event);
     for (const credential of this.crmCredentials) {
+      if (isTaskerEnabledForSalesforceCrm) {
+        if (!event.uid) {
+          console.error(
+            `Missing bookingId when scheduling CRM event creation on event type ${event?.eventTypeId}`
+          );
+          continue;
+        }
+
+        await CRMScheduler.createEvent({ bookingUid: event.uid });
```

Comment:
Since this is inside a credentials loop, wouldn't this create an event per each credential? How to do we know to which credential create with only the  bookingUid?

Reply:
Is it possible to have two CRM-type credentials installed and due to this get this scheduled two times? Resulting in duplicate events for both credentials? Do we have some kind of Idempotency logic in the createCRMEvent task?

Reply:
The credentials are passed to the `EventManager` when it's initialized. The credentials fetched via `getAllCredentials` which filters for only enabled CRM credentials on the event type.

Reply:
But even then, let's say I had both salesforce and hubspot installed (which both are CRM apps). This will generate duplicate data for both of them.

Reply:
You can have multiple CRM apps installed but only enabled CRM credentials for that specific event type are passed to the EventManager and iterated through in this loop.

The scenario you're describing did happen a long time ago but I refactored it in 2024 so we weren't iterating over every installed CRM app.